src/model.ts

```diff
@@ -7,67 +7,124 @@ import { IContentFieldFilter, IContentSearchRequest } from '@esri/hub-search';
 import { PassThrough } from 'stream';
 import { PagingStream } from './paging-stream';
 import { getBatchedStreams } from './helpers/get-batched-streams';
-import { fetchSite, getHubApiUrl, getPortalApiUrl, hubApiRequest, RemoteServerError } from '@esri/hub-common';
+import { fetchSite, getHubApiUrl, getPortalApiUrl, hubApiRequest, IModel, RemoteServerError } from '@esri/hub-common';
+
+const REQUIRED_FIELDS = [
+  'id', // used for the dataset landing page URL
+  'type', // used for the dataset landing page URL
+  'slug', // used for the dataset landing page URL
+  'access', // used for detecting proxied csv's
+  'size', // used for detecting proxied csv's
+  'licenseInfo', // required for license resolution
+  'structuredLicense', // required for license resolution
+];
+
+// additional fields due to dataset enrichment
+const ADDON_FIELDS = [
+  'hubLandingPage',
+  'accessUrlCSV',
+  'isLayer',
+  'accessUrlKML',
+  'accessUrlShapeFile',
+  'accessUrlWFS',
+  'accessUrlWMS',
+  'accessUrlGeoJSON',
+  'license'
+];
+
+type HubApiRequest = {
+  hostname: string,
+  res?: {
+    locals: {
+      searchRequestBody?: IContentSearchRequest,
+      siteModel?: IModel,
+      arcgisPortal?: string
+    }
+  }
+  query?: {
+    limit?: number
+  }
+}
 
 export class HubApiModel {
 
-  async getStream (request: Request) {
-    const searchRequest: IContentSearchRequest = request.res.locals.searchRequest;
-    this.preprocessSearchRequest(searchRequest);
+  async getStream(request: Request) {
+    const { hostname, res: { locals: { searchRequestBody, siteModel, arcgisPortal } }, query: { limit } }: HubApiRequest = request;
+    const hubSiteModel = siteModel || await this.fetchSiteModel(hostname, this.getRequestOptions(arcgisPortal));
+
+    if (!hubSiteModel) {
+      throw Error('Unable to fetch Hub site model');
+    }
```

Comment:
Is this necessary?  Under what cases does the method return undefined?

Reply:
I was getting undefined with a personal hubsite I tested. Don't see any reason why it would other than it didn't have any dataset. Now, it seems to be working fine. I have removed the check.